### DISCUSSION TITLE
what is the easiest way to be to upload data

### DISCUSSION THREAD

**Discussion by atherdon** [atherdon/dashb#94] [General]
Окей, мы сделали уже много подготовительной работы для вывода информации.
Выглядит хорошо, и для начала - этого достаточно чтобы показать MVP другим редакторам.

При этом - когда мы сделаем уже базовые страницы и редактора начнут использовать наш dashboard чаще чем основной - каждую неделю нужно будет обновлять данные.

Я нехочу делать uploader для csv/json файла. я хочу чтото более простое. например я сам смогу заливать csv/json в мастер ветку. при автоматическом деплое данные будут автоматически обновляться.

Я думаю что для начала нам нужно прийти к аккому то общему знаметанелю и понять, как лучше это сделать.

я также не думаю что нам нужен один большой и сложный json файл.

мы можем завести папку, для статических массивов. и определенный файл будет точно также отвечать за необходимые для нас данные. а я например смогу попросить другого редактора вводить данные в excel file.

понимаю, очень неудобно и глупо так делать - но если это ускорит наш процесс - тогда

**Comment by atherdon** [comment]
< Почему не сделать кластер на монгодб?

@master-bogdan можете развернуть идею?

**Comment by vadim9999** [comment]
Create a validator for all files in the data folder. Check all fields in an array if t exist or not. Add check to package.json in scripts and run it on build. it will display in github actions if build crashed.

**Comment by atherdon** [comment]
still a file.
but at some point we can build another separate project that will be just CRUD forms for population of our graphql backend
https://www.youtube.com/watch?v=Ovg9CYwWFBM&ab_channel=ReactEurope
https://github.com/rjsf-team/react-jsonschema-form
